fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java

```diff
@@ -350,17 +349,13 @@ public void dropTable(DropTableStmt stmt) {
         }
 
         icebergCatalog.dropTable(stmt.getDbName(), stmt.getTableName(), stmt.isForceDrop());
-        tables.remove(TableIdentifier.of(stmt.getDbName(), stmt.getTableName()));
         StatisticUtils.dropStatisticsAfterDropTable(icebergTable);
         asyncRefreshOthersFeMetadataCache(stmt.getDbName(), stmt.getTableName());
     }
 
     @Override
     public Table getTable(String dbName, String tblName) {
         TableIdentifier identifier = TableIdentifier.of(dbName, tblName);
-        if (tables.containsKey(identifier)) {
```

Comment:
if you remove this, does the table's query level cache still work?

Reply:
yes, we have cached the iceberg sdk table.

Reply:
If we do not have query level cache, there may be different snaphot for same table in one query